### ISSUE TITLE
HTML::Canvas::To::PDF

### ISSUE THREAD

**Issue by AlexDaniel** [perl6/ecosystem-unbitrot#581]
Module [HTML::Canvas::To::PDF](https://modules.perl6.org/dist/HTML::Canvas::To::PDF) cannot be installed (`AlwaysFail`), perhaps it has some failing tests.

* Tickets are opened/closed in this repo automatically (though not immediately).
* If you can **install the module without any problems**, add `works for me` label, leave a comment saying that it works for you and mention any details that you feel are important.
* If it **needs a native library**, put `native dependency` label, describe what you did to install it and ensure that same instructions are present in the README file of the module (otherwise submit a pull request). Also try to update [this wiki page](https://github.com/perl6/ecosystem-unbitrot/wiki).
* If the **module is broken**, try to fix it and send a PR. Add `PR sent` label.
* If there is a **problem in one of the dependencies**, add `failing dependency` label and write a comment explaining the situation. Feel free to work on the corresponding ticket for the failing dependency.
* It is a good idea to **assign yourself** to this ticket if you're working on it (to make sure two or more people are not working on the same ticket at the same time).
* **Once you are done**, [search for a next ticket](https://github.com/perl6/ecosystem-unbitrot/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+-label%3A%22PR+sent%22+-label%3A%22deprecated+module%22+-label%3A%22native+dependency%22).

If you can't self-assign yourself or attach a label, please let us know on [#perl6 channel on freenode](https://perl6.org/irc) or just leave a comment here. We will try to give you privileges as fast as possible.

Output:

````````````````````````````````````````````````````````
===> Searching for: HTML::Canvas::To::PDF
===> Found: HTML::Canvas::To::PDF:ver<0.0.2>:auth<github:p6-pdf> [via Zef::Repository::Ecosystems<cpan>]
===> Fetching [OK]: HTML::Canvas::To::PDF:ver<0.0.2>:auth<github:p6-pdf> to /home/alex/Blin/data/zef-data/tmp/1551239813.311.3085/HTML-Canvas-To-PDF-0.0.2.tar.gz
===> Extraction [OK]: HTML::Canvas::To::PDF to /home/alex/Blin/data/zef-data/store/HTML-Canvas-To-PDF-0.0.2.tar.gz
===> Testing: HTML::Canvas::To::PDF:ver<0.0.2>:auth<github:p6-pdf>
t/00-readme.t ............... ok
t/render-pdf-basic.t ........ ok
Unable to parse the pattern
unable to resolve font-name: courier:weight=bold
  in block  at /home/alex/Blin/installed/HTML::Canvas_0.0.7/sources/79297E3963C15A65CEB56DB09AEE597FC7B66D1C (HTML::Canvas::To::Cairo) line 37
falling back to mono-spaced font: /home/alex/Blin/installed/HTML::Canvas_0.0.7/resources/934B7A64D2B6806A6592BC9BE2BE532109A6A66D.ttf
  in block  at /home/alex/Blin/installed/HTML::Canvas_0.0.7/sources/79297E3963C15A65CEB56DB09AEE597FC7B66D1C (HTML::Canvas::To::Cairo) line 39
Unable to parse the pattern
stopped on page 1: unable to resolve font-name: courier:weight=bold
  in block  at t/render-pdf-test-sheets.t line 40
# Failed test 'completion of page 1'
# at t/render-pdf-test-sheets.t line 50
Unable to parse the pattern
unable to resolve font-name: arial:weight=bold
  in block  at /home/alex/Blin/installed/HTML::Canvas_0.0.7/sources/79297E3963C15A65CEB56DB09AEE597FC7B66D1C (HTML::Canvas::To::Cairo) line 37
falling back to mono-spaced font: /home/alex/Blin/installed/HTML::Canvas_0.0.7/resources/934B7A64D2B6806A6592BC9BE2BE532109A6A66D.ttf
  in block  at /home/alex/Blin/installed/HTML::Canvas_0.0.7/sources/79297E3963C15A65CEB56DB09AEE597FC7B66D1C (HTML::Canvas::To::Cairo) line 39
Unable to parse the pattern
stopped on page 2: unable to resolve font-name: arial:weight=bold
  in block  at t/render-pdf-test-sheets.t line 40
# Failed test 'completion of page 2'
# at t/render-pdf-test-sheets.t line 50
Unable to parse the pattern
unable to resolve font-name: courier:weight=bold
  in block  at /home/alex/Blin/installed/HTML::Canvas_0.0.7/sources/79297E3963C15A65CEB56DB09AEE597FC7B66D1C (HTML::Canvas::To::Cairo) line 37
falling back to mono-spaced font: /home/alex/Blin/installed/HTML::Canvas_0.0.7/resources/934B7A64D2B6806A6592BC9BE2BE532109A6A66D.ttf
  in block  at /home/alex/Blin/installed/HTML::Canvas_0.0.7/sources/79297E3963C15A65CEB56DB09AEE597FC7B66D1C (HTML::Canvas::To::Cairo) line 39
Unable to parse the pattern
unable to resolve font-name: courier:weight=bold
  in block  at /home/alex/Blin/installed/HTML::Canvas_0.0.7/sources/79297E3963C15A65CEB56DB09AEE597FC7B66D1C (HTML::Canvas::To::Cairo) line 37
falling back to mono-spaced font: /home/alex/Blin/installed/HTML::Canvas_0.0.7/resources/934B7A64D2B6806A6592BC9BE2BE532109A6A66D.ttf
  in block  at /home/alex/Blin/installed/HTML::Canvas_0.0.7/sources/79297E3963C15A65CEB56DB09AEE597FC7B66D1C (HTML::Canvas::To::Cairo) line 39
Unable to parse the pattern
stopped on page 8: unable to resolve font-name: courier:weight=bold
  in block  at t/render-pdf-test-sheets.t line 40
# Failed test 'completion of page 8'
# at t/render-pdf-test-sheets.t line 50
# Failed test 'pdf.save-as'
# at t/render-pdf-test-sheets.t line 682
# 'q' (Save) unmatched by closing 'Q' (Restore) at end of content stream
# Looks like you failed 4 tests of 13
t/render-pdf-test-sheets.t .. 
Dubious, test returned 4 (wstat 1024, 0x400)
Failed 4/13 subtests 
t/transforms.t .............. ok

Test Summary Report
-------------------
t/render-pdf-test-sheets.t (Wstat: 1024 Tests: 13 Failed: 4)
  Failed tests:  1, 3, 9, 13
  Non-zero exit status: 4
Files=4, Tests=36, 57 wallclock secs ( 0.03 usr  0.01 sys + 67.62 cusr  5.08 csys = 72.74 CPU)
Result: FAIL
===> Testing [FAIL]: HTML::Canvas::To::PDF:ver<0.0.2>:auth<github:p6-pdf>
Failed to get passing tests, but continuing with --force-test
===> Installing: HTML::Canvas::To::PDF:ver<0.0.2>:auth<github:p6-pdf>
===> Install [OK] for HTML::Canvas::To::PDF:ver<0.0.2>:auth<github:p6-pdf>

````````````````````````````````````````````````````````

Ping @p6-pdf

**Comment by dwarring** [comment]
Ah, similar to PDF::Canvas. Need to add a fallback font as a resource for testing purposes.

**Comment by dwarring** [comment]
Font fallback logic fixed in HTML::Canvas::To::PDF 0.0.3. Uploaded to CPAN.
